src/System.Private.CoreLib/shared/System/Runtime/MemoryFailPoint.cs

```diff
@@ -307,16 +310,17 @@ LastKnownFreeAddressSpace < (long)segmentSize)
 
             RuntimeHelpers.PrepareConstrainedRegions();
 
-            Interlocked.Add(ref s_failPointReservedMemory, (long)size);
+            AddMemoryFailPointReservation((long)size);
             _mustSubtractReservation = true;
 #endif
         }
 
+#if !FEATURE_PAL
```

Comment:
This won't work well for CoreRT and Mono on Unix.

Neither CoreRT or Mono have `FEATURE_PAL` defined. This will compile and try to call the Windows APIs on Unix that do not exist on Unix.

Reply:
Well, technically it was already broken on CoreRT. I can split it into two files (MemoryFailPoint.cs and MemoryFailPoint.Windows.cs) and provide a dummy implementation for non-Windows systems (MemoryFailPoint.Dummy.cs) until a proper implementation is created.